src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -146,11 +146,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
```

Comment:
I think that it'd be better to have a `TaskCompletionSource` and use `SetException`.